protocol/ibd/selected_tip.go

```diff
@@ -0,0 +1,133 @@
+package ibd
+
+import (
+	"errors"
+	"github.com/kaspanet/kaspad/blockdag"
+	"github.com/kaspanet/kaspad/netadapter/router"
+	"github.com/kaspanet/kaspad/protocol/common"
+	peerpkg "github.com/kaspanet/kaspad/protocol/peer"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/kaspanet/kaspad/wire"
+	"time"
+)
+
+const minDurationToRequestSelectedTips = time.Minute
+
+func requestSelectedTipsIfRequired(dag *blockdag.BlockDAG) error {
+	if hasRecentlyReceivedBlock(dag) {
+		return nil
+	}
+	return requestSelectedTips(dag)
+}
+
+func hasRecentlyReceivedBlock(dag *blockdag.BlockDAG) bool {
+	return dag.Now().Sub(dag.SelectedTipHeader().Timestamp) > minDurationToRequestSelectedTips
+}
+
+func requestSelectedTips(dag *blockdag.BlockDAG) error {
+	for _, peer := range peerpkg.ReadyPeers() {
+		peer.RequestSelectedTipIfRequired()
```

Comment:
I might be missing something, but I couldn't find the place where these selectedTips are being received.

Reply:
1. `RequestSelectedTip` waits on `peer.selectedTipRequestChan`
2. `RequestSelectedTipIfRequired` does: `peer.selectedTipRequestChan <- struct{}{}`
3. In the end of `RequestSelectedTip`: `peer.SetSelectedTipHash(peerSelectedTipHash)`